### ISSUE TITLE
 NU_LOG_LEVEL default should be higher than the current INFO (20)

### ISSUE THREAD

**Issue by bobhy** [nushell/nushell#8823]
### Related problem

Run a command that is assumed to be liberally instrumented with `log` commands of "correctly" chosen severity levels.

Expect:
To not see any messages from the command if everything worked.

Observe:
INFO level messages in addition to the expected result

### Describe the solution you'd like

Make WARN the default NU_LOG_LEVEL.  So you must explicitly NU_LOG_LEVEL=20 to see info messages.  This would be equivalent to invoking with a "verbose" flag.

### Describe alternatives you've considered

We could consider adding a NOTICE log severity for more consistency with `syslog`, and, if we did, *that* should be the default.

### Additional context and details

Per [unix syslog](https://en.wikipedia.org/wiki/Syslog) and long-standing tradition, Info level messages "confirm normal operation", that is, are similar to the additional messages you get when you run a unix command with "verbose" messages enabled -- everything's OK, but here's some additional detail about the operation.

I understand I could set a value for NU_LOG_LEVEL in my personal `env.nu` and override the hard-coded default, but I recommend we make the change in std log to so script developers can assume Nushell environment supports the traditional interpretation of log severity levels.

**Comment by rgwood** [comment]
Backing up a bit; why are we using `NU_LOG_LEVEL` at all? We already have a `--log-level` flag on `nu`; I don't think we should have 2 ways to set the log level.

**Comment by amtoine** [comment]
@rgwood
> Backing up a bit; why are we using `NU_LOG_LEVEL` at all? We already have a `--log-level` flag on `nu`; I don't think we should have 2 ways to set the log level.
>
i think the `std log` module has been written to log `nushell` scripts from inside nushell itself.
this is why it's using the `NU_LOG_LEVEL` environment variable, to be able to change the level when running scripts from a single `nushell`

e.g. what happens if you forgot to set `--log-level` properly before running `nushell`? 
=> i'd say it's ok to have these two ways of setting the log level because they are not the same log levels!

on the other hand, maybe `nu --log-level` should give the appropriate default value to `NU_LOG_LEVEL` when starting the shell?

**Comment by amtoine** [comment]
@bobhy 

you're request is about changing the default `NU_LOG_LEVEL` from `INFO` to `WARN`, is that right? :yum:

**Comment by bobhy** [comment]
@amtoine Yes, change to WARN.

**Comment by amtoine** [comment]
> @amtoine Yes, change to WARN.
>
gotcha :ok_hand: 

that looks sensible to me :+1:

**Comment by rgwood** [comment]
> on the other hand, maybe `nu --log-level` should give the appropriate default value to `NU_LOG_LEVEL` when starting the shell?

Yeah, that could work or we could deprecate `--log-level` in favour of `NU_LOG_LEVEL`. I don't have a strong preference for either, but I do think it's worth making logging behaviour consistent across Rust code and the Nu stdlib. Not a super high priority though.

**Comment by fdncred** [comment]
I'm personally fine with some logging mechanism explicitly for the stdlib like NU_LOG_LEVEL, but maybe it should be named NUSTD_LOG_LEVEL, or something indicating that it's explicitly for the stdlib. 

I'd really not want to see --log-level used for nushell and stdlib because all the messages would be intermingled and you would have a hard time separating out which log entries are from nushell and which are from the stdlib. If I'm debugging something I wrote in the stdlib, I'd want to be able to see only those log entries.

**Comment by rgwood** [comment]
> you would have a hard time separating out which log entries are from nushell and which are from the stdlib

Dumb question: why would you need to do that?

I think of the stdlib as, essentially, an implementation detail; the presence of a command in the stdlib means we decided to write the command in Nu instead of Rust. Language details aside, the user+developer experience should be roughly the same in both.

**Comment by fdncred** [comment]
> Dumb question: why would you need to do that?

Maybe you haven't used log-level that much? The problem is that when I'm debugging some thing in the stdlib, I don't want to see the 1000 lines of logs from other parts of nushell. I'd like to keep things easier for the developer rather than spitting out a mess of totally unrelated logs. 

Let's keep them separate.

**Comment by rgwood** [comment]
It seems like you're bothered by existing logging being too verbose. I don't think that giving the stdlib a different logging mechanism is a good way to address that.

In particular, it seems bad if users+devs have to know the answer to "is this command written in Rust or in Nu?" before turning on logging.

**Comment by fdncred** [comment]
> It seems like you're bothered by existing logging being too verbose.

Nope. The verbosity is what you need for figuring out parser issues. 

It's not what you need debugging the stdlib.

**Comment by presidento** [comment]
In other programs it is common to display some basic useful information and allow the user to both make it more verbose or quiet. For example `git`. If you want to use it in `cron` you can apply `-q`, although the default setting for interactive usage is some info-ish level.

**Comment by bobhy** [comment]
To focus these ideas a bit, we've decided to retain `NU_LOG_LEVEL` and initialize it to `--log-level`.  
For me, the key missing feature is ability to specify different log levels for different components (STDLIB is just one example, maybe each crate and script module is the right level of componentization?)  In any case, this request then would become to specify WARN as the default log level for script components. (again, STDLIB is just one such).

**Comment by Larandar** [comment]
Just to weight in on an old issue, as a user I searched for a few (dozens of) minutes before finding this thread and realize that `--log-level` is deprecated, and affect both performance informations and "shell logs" that are displayed with different formats:

<img width="1440" alt="image" src="https://github.com/nushell/nushell/assets/906021/44623086-a040-467c-be78-c9cfa8f97a2a">

But on the contrary `NU_LOG_LEVEL` only work for scripts (which is what I wanted, but tried --verbose --log-<TAB> ect before).

So if someone is looking for that glad I helped :)

**Comment by Larandar** [comment]
Brain, words hard 😅 corrected sorry

**Comment by Kissaki** [comment]
To first summarize the current state (tested):

`nu --log-level` sets the log level for Nushell commands.

`$env.NU_LOG_LEVEL` sets the log level for `std/log`.

They do not influence each other.

---

This ticket asks to reduce the default `$env.NU_LOG_LEVEL` from INFO to WARN.

While that seemingly makes sense in a shell context, outside of that - in a broader programming / software development context - it does not. There you probably expect INFO as the default.

In Nushell, this only takes effect when 1. including `std/log` yourself, or 2. using modules that use it.

A default Nushell installation has neither, and will consequently not print any log messages.

A user can set a different default log level in their `env.nu` or `config.nu`.

It may be preferable to show more instead of less by default in order not to hide significant or guiding information.

---

With these points in mind, I feel like INFO may be a slightly better default, even in a shell environment.

In a default installation, no logs show up. When a user includes modules or scripts that do use `std/log`, and they desire no INFO, they can change their default log level environment variable configuration.

**Comment by Kissaki** [comment]
I wouldn't have an issue with a default WARN either, though, given the primary shell and scripting target.

**Comment by Kissaki** [comment]
I saw now that #8829 discussion went into a direction of full-fledged logging framework with components. Either way, it's stale since 2023, and the current state is still the two separate logging mechanisms, so the comments above still make sense for the current state.
